DLQ enabling is not working with 'http-request'

I'm using the `http-request 3.1.1` app in `SCDF 2.9.1` and I want the Kafka message to be stored in a DLQ in case the HTTP service invoked by `http-request` is not responding.
The idea is to resubmit the message when the HTTP service is back up.

I'm using this configuration :
```
--spring.cloud.stream.kafka.default.consumer.enableDlq=true
--spring.cloud.stream.kafka.default.consumer.dlqName=errorHttpRequest
```

I have this error :
```
c.f.c.c.BeanFactoryAwareFunctionRegistry : Failed to invoke function 'httpRequestFunction'
2021-12-21T11:48:40.076935527+01:00 stdout F 
2021-12-21T11:48:40.076942067+01:00 stdout F org.springframework.web.reactive.function.client.WebClientRequestException: finishConnect(..) failed: Connection refused: /<IP>:8085; nested exception is io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /<IP>:8085
```
Which is normal but nothing in the DLQ ...

Note that this was working fine with the previous app `httpclient`

Thank you for the explanation.
Here is the full stack trace :
```
2021-12-21T15:48:08.886695136+01:00 stdout F 2021-12-21 14:48:08.882 ERROR [http-request-processor,91548655281a3f62,59606d272eb2fb73] 1 --- [or-http-epoll-2] c.f.c.c.BeanFactoryAwareFunctionRegistry : Failed to invoke function 'httpRequestFunction'
2021-12-21T15:48:08.886709206+01:00 stdout F 
2021-12-21T15:48:08.886715458+01:00 stdout F org.springframework.web.reactive.function.client.WebClientRequestException: finishConnect(..) failed: Connection refused: /<IP>:8085; nested exception is io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /<IP>:8085
2021-12-21T15:48:08.886721814+01:00 stdout F 	at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:141) ~[spring-webflux-5.3.10.jar:5.3.10]
2021-12-21T15:48:08.886726604+01:00 stdout F 	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
2021-12-21T15:48:08.886729997+01:00 stdout F Error has been observed at the following site(s):
2021-12-21T15:48:08.886733002+01:00 stdout F 	|_ checkpoint ? Request to POST http://<IP>:8085/send [DefaultWebClient]
2021-12-21T15:48:08.886736074+01:00 stdout F Stack trace:
2021-12-21T15:48:08.886739101+01:00 stdout F 		at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:141) ~[spring-webflux-5.3.10.jar:5.3.10]
2021-12-21T15:48:08.886742104+01:00 stdout F 		at reactor.core.publisher.MonoErrorSupplied.subscribe(MonoErrorSupplied.java:55) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886745963+01:00 stdout F 		at reactor.core.publisher.Mono.subscribe(Mono.java:4361) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886749356+01:00 stdout F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886755417+01:00 stdout F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886759589+01:00 stdout F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886762545+01:00 stdout F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886765586+01:00 stdout F 		at reactor.core.publisher.MonoNext$NextSubscriber.onError(MonoNext.java:93) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886768531+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.886771655+01:00 stdout F 		at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onError(MonoFlatMapMany.java:204) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886774577+01:00 stdout F 		at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886788896+01:00 stdout F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.whenError(FluxRetryWhen.java:225) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886791875+01:00 stdout F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenOtherSubscriber.onError(FluxRetryWhen.java:274) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886794483+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.886797055+01:00 stdout F 		at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.drain(FluxConcatMap.java:414) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886799798+01:00 stdout F 		at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.onNext(FluxConcatMap.java:251) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886802746+01:00 stdout F 		at reactor.core.publisher.EmitterProcessor.drain(EmitterProcessor.java:491) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886805286+01:00 stdout F 		at reactor.core.publisher.EmitterProcessor.tryEmitNext(EmitterProcessor.java:299) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886807851+01:00 stdout F 		at reactor.core.publisher.SinkManySerialized.tryEmitNext(SinkManySerialized.java:100) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886810598+01:00 stdout F 		at reactor.core.publisher.InternalManySink.emitNext(InternalManySink.java:27) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886813141+01:00 stdout F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.onError(FluxRetryWhen.java:190) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886815725+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.886818406+01:00 stdout F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.88682094+01:00 stdout F 		at reactor.netty.http.client.HttpClientConnect$MonoHttpConnect$ClientTransportSubscriber.onError(HttpClientConnect.java:304) ~[reactor-netty-http-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.886828564+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.886834384+01:00 stdout F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886837108+01:00 stdout F 		at reactor.netty.resources.DefaultPooledConnectionProvider$DisposableAcquire.onError(DefaultPooledConnectionProvider.java:172) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.886839702+01:00 stdout F 		at reactor.netty.internal.shaded.reactor.pool.AbstractPool$Borrower.fail(AbstractPool.java:445) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.886842295+01:00 stdout F 		at reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.lambda$drainLoop$9(SimpleDequePool.java:400) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.886846061+01:00 stdout F 		at reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.onError(FluxDoOnEach.java:186) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886848591+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.886851177+01:00 stdout F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886853726+01:00 stdout F 		at reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnectionAllocator$PooledConnectionInitializer.onError(DefaultPooledConnectionProvider.java:554) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.886859533+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.886863761+01:00 stdout F 		at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886866389+01:00 stdout F 		at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886868948+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.886871544+01:00 stdout F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886874129+01:00 stdout F 		at reactor.core.publisher.Operators.error(Operators.java:198) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886876706+01:00 stdout F 		at reactor.core.publisher.MonoError.subscribe(MonoError.java:53) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886879449+01:00 stdout F 		at reactor.core.publisher.Mono.subscribe(Mono.java:4361) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886882082+01:00 stdout F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.886884637+01:00 stdout F 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.tryFailure(TransportConnector.java:517) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.886887203+01:00 stdout F 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.setFailure(TransportConnector.java:471) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.886889752+01:00 stdout F 		at reactor.netty.transport.TransportConnector.lambda$doConnect$7(TransportConnector.java:206) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.886892317+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886894858+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.88689742+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.88690011+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.88690269+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886905301+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:609) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886907891+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886910474+01:00 stdout F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.fulfillConnectPromise(AbstractEpollChannel.java:674) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.886914462+01:00 stdout F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:693) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.886917285+01:00 stdout F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.88692254+01:00 stdout F 		at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:470) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.886925294+01:00 stdout F 		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.886927904+01:00 stdout F 		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886930457+01:00 stdout F 		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886933998+01:00 stdout F 		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886936571+01:00 stdout F 		at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
2021-12-21T15:48:08.886939438+01:00 stdout F Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /<IP>:8085
2021-12-21T15:48:08.886941993+01:00 stdout F Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
2021-12-21T15:48:08.886944744+01:00 stdout F 	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886947464+01:00 stdout F 	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886950028+01:00 stdout F 	at io.netty.channel.unix.Socket.finishConnect(Socket.java:320) ~[netty-transport-native-unix-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886953286+01:00 stdout F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.886955896+01:00 stdout F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.886958475+01:00 stdout F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.886961084+01:00 stdout F 	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:470) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.886963628+01:00 stdout F 	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.886966183+01:00 stdout F 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.88696874+01:00 stdout F 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886971319+01:00 stdout F 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.886973903+01:00 stdout F 	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
2021-12-21T15:48:08.886976633+01:00 stdout F 
2021-12-21T15:48:08.891768119+01:00 stdout F 2021-12-21 14:48:08.886 ERROR [http-request-processor,91548655281a3f62,59606d272eb2fb73] 1 --- [or-http-epoll-2] c.f.c.c.BeanFactoryAwareFunctionRegistry : Failed to convert output
2021-12-21T15:48:08.891798427+01:00 stdout F 
2021-12-21T15:48:08.891807295+01:00 stdout F org.springframework.web.reactive.function.client.WebClientRequestException: finishConnect(..) failed: Connection refused: /<IP>:8085; nested exception is io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /<IP>:8085
2021-12-21T15:48:08.891820412+01:00 stdout F 	at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:141) ~[spring-webflux-5.3.10.jar:5.3.10]
2021-12-21T15:48:08.891823958+01:00 stdout F 	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
2021-12-21T15:48:08.89182792+01:00 stdout F Error has been observed at the following site(s):
2021-12-21T15:48:08.891830815+01:00 stdout F 	|_ checkpoint ? Request to POST http://<IP>:8085/send [DefaultWebClient]
2021-12-21T15:48:08.89183456+01:00 stdout F Stack trace:
2021-12-21T15:48:08.8918378+01:00 stdout F 		at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:141) ~[spring-webflux-5.3.10.jar:5.3.10]
2021-12-21T15:48:08.891840638+01:00 stdout F 		at reactor.core.publisher.MonoErrorSupplied.subscribe(MonoErrorSupplied.java:55) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891843785+01:00 stdout F 		at reactor.core.publisher.Mono.subscribe(Mono.java:4361) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891846794+01:00 stdout F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891852219+01:00 stdout F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891855843+01:00 stdout F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891858405+01:00 stdout F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891860943+01:00 stdout F 		at reactor.core.publisher.MonoNext$NextSubscriber.onError(MonoNext.java:93) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891863523+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.891866075+01:00 stdout F 		at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onError(MonoFlatMapMany.java:204) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891868961+01:00 stdout F 		at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891871723+01:00 stdout F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.whenError(FluxRetryWhen.java:225) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891874568+01:00 stdout F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenOtherSubscriber.onError(FluxRetryWhen.java:274) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891877122+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.891879685+01:00 stdout F 		at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.drain(FluxConcatMap.java:414) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891882176+01:00 stdout F 		at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.onNext(FluxConcatMap.java:251) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891884725+01:00 stdout F 		at reactor.core.publisher.EmitterProcessor.drain(EmitterProcessor.java:491) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891887249+01:00 stdout F 		at reactor.core.publisher.EmitterProcessor.tryEmitNext(EmitterProcessor.java:299) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891889775+01:00 stdout F 		at reactor.core.publisher.SinkManySerialized.tryEmitNext(SinkManySerialized.java:100) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891896095+01:00 stdout F 		at reactor.core.publisher.InternalManySink.emitNext(InternalManySink.java:27) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891898856+01:00 stdout F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.onError(FluxRetryWhen.java:190) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891901535+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.891904114+01:00 stdout F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891906673+01:00 stdout F 		at reactor.netty.http.client.HttpClientConnect$MonoHttpConnect$ClientTransportSubscriber.onError(HttpClientConnect.java:304) ~[reactor-netty-http-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.891914955+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.891917965+01:00 stdout F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891920515+01:00 stdout F 		at reactor.netty.resources.DefaultPooledConnectionProvider$DisposableAcquire.onError(DefaultPooledConnectionProvider.java:172) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.891923058+01:00 stdout F 		at reactor.netty.internal.shaded.reactor.pool.AbstractPool$Borrower.fail(AbstractPool.java:445) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.891925624+01:00 stdout F 		at reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.lambda$drainLoop$9(SimpleDequePool.java:400) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.891929983+01:00 stdout F 		at reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.onError(FluxDoOnEach.java:186) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891932586+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.891935117+01:00 stdout F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.891966879+01:00 stdout F 		at reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnectionAllocator$PooledConnectionInitializer.onError(DefaultPooledConnectionProvider.java:554) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.892048126+01:00 stderr F org.springframework.web.reactive.function.client.WebClientRequestException: finishConnect(..) failed: Connection refused: /<IP>:8085; nested exception is io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /<IP>:8085
2021-12-21T15:48:08.892055776+01:00 stderr F 	at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:141)
2021-12-21T15:48:08.89205991+01:00 stderr F 	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
2021-12-21T15:48:08.892112261+01:00 stderr F Error has been observed at the following site(s):
2021-12-21T15:48:08.892117179+01:00 stderr F 	|_ checkpoint ? Request to POST http://<IP>:8085/send [DefaultWebClient]
2021-12-21T15:48:08.892120317+01:00 stderr F Stack trace:
2021-12-21T15:48:08.892123074+01:00 stderr F 		at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:141)
2021-12-21T15:48:08.892128045+01:00 stderr F 		at reactor.core.publisher.MonoErrorSupplied.subscribe(MonoErrorSupplied.java:55)
2021-12-21T15:48:08.892132111+01:00 stderr F 		at reactor.core.publisher.Mono.subscribe(Mono.java:4361)
2021-12-21T15:48:08.892140713+01:00 stderr F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103)
2021-12-21T15:48:08.892144172+01:00 stderr F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222)
2021-12-21T15:48:08.892146785+01:00 stderr F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222)
2021-12-21T15:48:08.892149322+01:00 stderr F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222)
2021-12-21T15:48:08.892151884+01:00 stderr F 		at reactor.core.publisher.MonoNext$NextSubscriber.onError(MonoNext.java:93)
2021-12-21T15:48:08.892154436+01:00 stderr F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95)
2021-12-21T15:48:08.892158527+01:00 stderr F 		at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onError(MonoFlatMapMany.java:204)
2021-12-21T15:48:08.892161174+01:00 stderr F 		at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124)
2021-12-21T15:48:08.89216386+01:00 stderr F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.whenError(FluxRetryWhen.java:225)
2021-12-21T15:48:08.89216644+01:00 stderr F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenOtherSubscriber.onError(FluxRetryWhen.java:274)
2021-12-21T15:48:08.89216915+01:00 stderr F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95)
2021-12-21T15:48:08.892171725+01:00 stderr F 		at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.drain(FluxConcatMap.java:414)
2021-12-21T15:48:08.8921743+01:00 stderr F 		at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.onNext(FluxConcatMap.java:251)
2021-12-21T15:48:08.892176947+01:00 stderr F 		at reactor.core.publisher.EmitterProcessor.drain(EmitterProcessor.java:491)
2021-12-21T15:48:08.892179501+01:00 stderr F 		at reactor.core.publisher.EmitterProcessor.tryEmitNext(EmitterProcessor.java:299)
2021-12-21T15:48:08.892183031+01:00 stderr F 		at reactor.core.publisher.SinkManySerialized.tryEmitNext(SinkManySerialized.java:100)
2021-12-21T15:48:08.892185797+01:00 stderr F 		at reactor.core.publisher.InternalManySink.emitNext(InternalManySink.java:27)
2021-12-21T15:48:08.892188412+01:00 stderr F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.onError(FluxRetryWhen.java:190)
2021-12-21T15:48:08.892191048+01:00 stderr F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95)
2021-12-21T15:48:08.892193592+01:00 stderr F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189)
2021-12-21T15:48:08.892196226+01:00 stderr F 		at reactor.netty.http.client.HttpClientConnect$MonoHttpConnect$ClientTransportSubscriber.onError(HttpClientConnect.java:304)
2021-12-21T15:48:08.89219876+01:00 stderr F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95)
2021-12-21T15:48:08.892201306+01:00 stderr F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189)
2021-12-21T15:48:08.892203881+01:00 stderr F 		at reactor.netty.resources.DefaultPooledConnectionProvider$DisposableAcquire.onError(DefaultPooledConnectionProvider.java:172)
2021-12-21T15:48:08.892206584+01:00 stderr F 		at reactor.netty.internal.shaded.reactor.pool.AbstractPool$Borrower.fail(AbstractPool.java:445)
2021-12-21T15:48:08.892209125+01:00 stderr F 		at reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.lambda$drainLoop$9(SimpleDequePool.java:400)
2021-12-21T15:48:08.892213065+01:00 stderr F 		at reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.onError(FluxDoOnEach.java:186)
2021-12-21T15:48:08.89221601+01:00 stderr F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95)
2021-12-21T15:48:08.89222171+01:00 stderr F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189)
2021-12-21T15:48:08.892224496+01:00 stderr F 		at reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnectionAllocator$PooledConnectionInitializer.onError(DefaultPooledConnectionProvider.java:554)
2021-12-21T15:48:08.892227711+01:00 stderr F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95)
2021-12-21T15:48:08.892261239+01:00 stderr F 		at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192)
2021-12-21T15:48:08.892265869+01:00 stderr F 		at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259)
2021-12-21T15:48:08.892268744+01:00 stderr F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95)
2021-12-21T15:48:08.8922716+01:00 stderr F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106)
2021-12-21T15:48:08.892274231+01:00 stderr F 		at reactor.core.publisher.Operators.error(Operators.java:198)
2021-12-21T15:48:08.892277674+01:00 stderr F 		at reactor.core.publisher.MonoError.subscribe(MonoError.java:53)
2021-12-21T15:48:08.892280299+01:00 stderr F 		at reactor.core.publisher.Mono.subscribe(Mono.java:4361)
2021-12-21T15:48:08.892282955+01:00 stderr F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103)
2021-12-21T15:48:08.89228561+01:00 stderr F 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.tryFailure(TransportConnector.java:517)
2021-12-21T15:48:08.892288324+01:00 stderr F 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.setFailure(TransportConnector.java:471)
2021-12-21T15:48:08.892290869+01:00 stderr F 		at reactor.netty.transport.TransportConnector.lambda$doConnect$7(TransportConnector.java:206)
2021-12-21T15:48:08.892293423+01:00 stderr F 		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
2021-12-21T15:48:08.89229597+01:00 stderr F 		at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571)
2021-12-21T15:48:08.8922999+01:00 stderr F 		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550)
2021-12-21T15:48:08.892302462+01:00 stderr F 		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491)
2021-12-21T15:48:08.892306746+01:00 stderr F 		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616)
2021-12-21T15:48:08.892309577+01:00 stderr F 		at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:609)
2021-12-21T15:48:08.892312162+01:00 stderr F 		at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117)
2021-12-21T15:48:08.892315417+01:00 stderr F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.fulfillConnectPromise(AbstractEpollChannel.java:674)
2021-12-21T15:48:08.892318591+01:00 stderr F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:693)
2021-12-21T15:48:08.892321203+01:00 stderr F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567)
2021-12-21T15:48:08.89232382+01:00 stderr F 		at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:470)
2021-12-21T15:48:08.892326436+01:00 stderr F 		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
2021-12-21T15:48:08.892329206+01:00 stderr F 		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
2021-12-21T15:48:08.892331758+01:00 stderr F 		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2021-12-21T15:48:08.892334462+01:00 stderr F 		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2021-12-21T15:48:08.892336986+01:00 stderr F 		at java.base/java.lang.Thread.run(Unknown Source)
2021-12-21T15:48:08.892345337+01:00 stderr F Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /<IP>:8085
2021-12-21T15:48:08.892348417+01:00 stderr F Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
2021-12-21T15:48:08.892351001+01:00 stderr F 	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155)
2021-12-21T15:48:08.892353917+01:00 stderr F 	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128)
2021-12-21T15:48:08.8923565+01:00 stderr F 	at io.netty.channel.unix.Socket.finishConnect(Socket.java:320)
2021-12-21T15:48:08.892359076+01:00 stderr F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710)
2021-12-21T15:48:08.892363999+01:00 stderr F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687)
2021-12-21T15:48:08.892366666+01:00 stderr F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567)
2021-12-21T15:48:08.892369277+01:00 stderr F 	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:470)
2021-12-21T15:48:08.892371946+01:00 stderr F 	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
2021-12-21T15:48:08.892374521+01:00 stderr F 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
2021-12-21T15:48:08.89237714+01:00 stderr F 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
2021-12-21T15:48:08.892379712+01:00 stderr F 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2021-12-21T15:48:08.89238229+01:00 stderr F 	at java.base/java.lang.Thread.run(Unknown Source)
2021-12-21T15:48:08.891973601+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.892453588+01:00 stdout F 		at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.892458266+01:00 stdout F 		at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.892461178+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.892464059+01:00 stdout F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.892466973+01:00 stdout F 		at reactor.core.publisher.Operators.error(Operators.java:198) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.892469667+01:00 stdout F 		at reactor.core.publisher.MonoError.subscribe(MonoError.java:53) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.892472677+01:00 stdout F 		at reactor.core.publisher.Mono.subscribe(Mono.java:4361) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.892475365+01:00 stdout F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.892478022+01:00 stdout F 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.tryFailure(TransportConnector.java:517) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.892480572+01:00 stdout F 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.setFailure(TransportConnector.java:471) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.892483147+01:00 stdout F 		at reactor.netty.transport.TransportConnector.lambda$doConnect$7(TransportConnector.java:206) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.89248568+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892494777+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892499475+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892503286+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892507158+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892511504+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:609) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892515245+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892519406+01:00 stdout F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.fulfillConnectPromise(AbstractEpollChannel.java:674) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.892531497+01:00 stdout F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:693) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.892535807+01:00 stdout F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.892538559+01:00 stdout F 		at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:470) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.892541514+01:00 stdout F 		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.89254735+01:00 stdout F 		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892549993+01:00 stdout F 		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892552578+01:00 stdout F 		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892555111+01:00 stdout F 		at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
2021-12-21T15:48:08.892557852+01:00 stdout F Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /<IP>:8085
2021-12-21T15:48:08.892561206+01:00 stdout F Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
2021-12-21T15:48:08.892563993+01:00 stdout F 	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892566559+01:00 stdout F 	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.89256912+01:00 stdout F 	at io.netty.channel.unix.Socket.finishConnect(Socket.java:320) ~[netty-transport-native-unix-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.89257169+01:00 stdout F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.892574261+01:00 stdout F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.89258054+01:00 stdout F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.892583281+01:00 stdout F 	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:470) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.892585953+01:00 stdout F 	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.892596951+01:00 stdout F 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892599585+01:00 stdout F 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.892602168+01:00 stdout F 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.89260471+01:00 stdout F 	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
2021-12-21T15:48:08.89260722+01:00 stdout F 
2021-12-21T15:48:08.892610212+01:00 stdout F 2021-12-21 14:48:08.888 ERROR [http-request-processor,91548655281a3f62,59606d272eb2fb73] 1 --- [or-http-epoll-2] onfiguration$FunctionToDestinationBinder : Failure was detected during execution of the reactive function 'httpRequestFunction'
2021-12-21T15:48:08.897181211+01:00 stdout F 2021-12-21 14:48:08.892 ERROR [http-request-processor,91548655281a3f62,59606d272eb2fb73] 1 --- [or-http-epoll-2] reactor.core.publisher.Operators         : Operator called default onErrorDropped
2021-12-21T15:48:08.897195625+01:00 stdout F 
2021-12-21T15:48:08.897202544+01:00 stdout F reactor.core.Exceptions$ErrorCallbackNotImplemented: org.springframework.web.reactive.function.client.WebClientRequestException: finishConnect(..) failed: Connection refused: /<IP>:8085; nested exception is io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /<IP>:8085
2021-12-21T15:48:08.897208223+01:00 stdout F Caused by: org.springframework.web.reactive.function.client.WebClientRequestException: finishConnect(..) failed: Connection refused: /<IP>:8085; nested exception is io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /<IP>:8085
2021-12-21T15:48:08.897212369+01:00 stdout F 	at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:141) ~[spring-webflux-5.3.10.jar:5.3.10]
2021-12-21T15:48:08.897215408+01:00 stdout F 	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
2021-12-21T15:48:08.897218459+01:00 stdout F Error has been observed at the following site(s):
2021-12-21T15:48:08.897221111+01:00 stdout F 	|_ checkpoint ? Request to POST http://<IP>:8085/send [DefaultWebClient]
2021-12-21T15:48:08.897224125+01:00 stdout F Stack trace:
2021-12-21T15:48:08.897226825+01:00 stdout F 		at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:141) ~[spring-webflux-5.3.10.jar:5.3.10]
2021-12-21T15:48:08.897229435+01:00 stdout F 		at reactor.core.publisher.MonoErrorSupplied.subscribe(MonoErrorSupplied.java:55) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897233296+01:00 stdout F 		at reactor.core.publisher.Mono.subscribe(Mono.java:4361) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.89723743+01:00 stdout F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897240732+01:00 stdout F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897251689+01:00 stdout F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897254548+01:00 stdout F 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897257111+01:00 stdout F 		at reactor.core.publisher.MonoNext$NextSubscriber.onError(MonoNext.java:93) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897262346+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.897264959+01:00 stdout F 		at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onError(MonoFlatMapMany.java:204) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897267584+01:00 stdout F 		at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897270193+01:00 stdout F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.whenError(FluxRetryWhen.java:225) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897272768+01:00 stdout F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenOtherSubscriber.onError(FluxRetryWhen.java:274) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897275339+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.897277913+01:00 stdout F 		at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.drain(FluxConcatMap.java:414) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897280457+01:00 stdout F 		at reactor.core.publisher.FluxConcatMap$ConcatMapImmediate.onNext(FluxConcatMap.java:251) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897283227+01:00 stdout F 		at reactor.core.publisher.EmitterProcessor.drain(EmitterProcessor.java:491) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897285786+01:00 stdout F 		at reactor.core.publisher.EmitterProcessor.tryEmitNext(EmitterProcessor.java:299) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897288324+01:00 stdout F 		at reactor.core.publisher.SinkManySerialized.tryEmitNext(SinkManySerialized.java:100) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897290872+01:00 stdout F 		at reactor.core.publisher.InternalManySink.emitNext(InternalManySink.java:27) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897293415+01:00 stdout F 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.onError(FluxRetryWhen.java:190) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897299408+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.89730233+01:00 stdout F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897305001+01:00 stdout F 		at reactor.netty.http.client.HttpClientConnect$MonoHttpConnect$ClientTransportSubscriber.onError(HttpClientConnect.java:304) ~[reactor-netty-http-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.897309069+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.897311673+01:00 stdout F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897333822+01:00 stdout F 		at reactor.netty.resources.DefaultPooledConnectionProvider$DisposableAcquire.onError(DefaultPooledConnectionProvider.java:172) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.897345912+01:00 stdout F 		at reactor.netty.internal.shaded.reactor.pool.AbstractPool$Borrower.fail(AbstractPool.java:445) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.897349273+01:00 stdout F 		at reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.lambda$drainLoop$9(SimpleDequePool.java:400) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.89735556+01:00 stdout F 		at reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.onError(FluxDoOnEach.java:186) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897358415+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.897362486+01:00 stdout F 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:189) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897365343+01:00 stdout F 		at reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnectionAllocator$PooledConnectionInitializer.onError(DefaultPooledConnectionProvider.java:554) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.897368924+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.897372624+01:00 stdout F 		at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897375412+01:00 stdout F 		at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897378111+01:00 stdout F 		at org.springframework.cloud.sleuth.instrument.reactor.ScopePassingSpanSubscriber.onError(ScopePassingSpanSubscriber.java:95) ~[spring-cloud-sleuth-instrumentation-3.0.4.jar:3.0.4]
2021-12-21T15:48:08.897380787+01:00 stdout F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.89738356+01:00 stdout F 		at reactor.core.publisher.Operators.error(Operators.java:198) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897386333+01:00 stdout F 		at reactor.core.publisher.MonoError.subscribe(MonoError.java:53) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897389036+01:00 stdout F 		at reactor.core.publisher.Mono.subscribe(Mono.java:4361) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897391638+01:00 stdout F 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.4.10.jar:3.4.10]
2021-12-21T15:48:08.897394223+01:00 stdout F 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.tryFailure(TransportConnector.java:517) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.897396821+01:00 stdout F 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.setFailure(TransportConnector.java:471) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.897399521+01:00 stdout F 		at reactor.netty.transport.TransportConnector.lambda$doConnect$7(TransportConnector.java:206) ~[reactor-netty-core-1.0.11.jar:1.0.11]
2021-12-21T15:48:08.897402147+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897404727+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897407342+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897412076+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897416896+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897422631+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:609) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897425856+01:00 stdout F 		at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.89742851+01:00 stdout F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.fulfillConnectPromise(AbstractEpollChannel.java:674) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.897431119+01:00 stdout F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:693) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.897433723+01:00 stdout F 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.897436328+01:00 stdout F 		at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:470) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.897439002+01:00 stdout F 		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.8974416+01:00 stdout F 		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.8974442+01:00 stdout F 		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897446865+01:00 stdout F 		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897449445+01:00 stdout F 		at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
2021-12-21T15:48:08.897452089+01:00 stdout F Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /<IP>:8085
2021-12-21T15:48:08.89745466+01:00 stdout F Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
2021-12-21T15:48:08.897457322+01:00 stdout F 	at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155) ~[netty-transport-native-unix-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.8974599+01:00 stdout F 	at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128) ~[netty-transport-native-unix-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897462629+01:00 stdout F 	at io.netty.channel.unix.Socket.finishConnect(Socket.java:320) ~[netty-transport-native-unix-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897465289+01:00 stdout F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.89746951+01:00 stdout F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.897472234+01:00 stdout F 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.897474828+01:00 stdout F 	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:470) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.897477376+01:00 stdout F 	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378) ~[netty-transport-native-epoll-4.1.68.Final-linux-x86_64.jar:4.1.68.Final]
2021-12-21T15:48:08.897485963+01:00 stdout F 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897491319+01:00 stdout F 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897493994+01:00 stdout F 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
2021-12-21T15:48:08.897496597+01:00 stdout F 	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
2021-12-21T15:48:08.897499084+01:00 stdout F 
```
If I understand what you say and what I read about functions, I have to configure DLQ for system errors et errorChannel for application errors if I want both. Is that correct ?

--- a/.github/workflows/ci-pr.yml
+++ b/.github/workflows/ci-pr.yml
@@ -10,15 +10,15 @@ jobs:
   check:
     runs-on: 'stream-ci-large'
     steps:
-      - name: 'Configure: checkout'
-        uses: actions/checkout@v3
-        with:
-          ref: 'main'
       - name: 'Configure: checkout stream-applications@${{ inputs.branch }}'
         uses: actions/checkout@v3
         with:
           ref: ${{ github.ref }}
-          path: 'stream-applications'
+      - name: 'Configure: checkout'
+        uses: actions/checkout@v3
+        with:
+          ref: 'main'
+          path: 'main-build'
       - name: 'Configure: Ensure scripts are executable'
         shell: bash
         run: find . -type f -name "*.sh" -exec chmod a+x '{}' \;
@@ -49,8 +49,8 @@ jobs:
       - name: 'Action: build initial dependencies'
         shell: bash
         run: |
-          ROOT_DIR=$(realpath $PWD)
-          MAVEN_OPT="-U -B -T 1C -s $ROOT_DIR/.settings.xml ${{ inputs.verbose && '--debug' || '' }}"
+          BUILD_DIR=./main-build
+          MAVEN_OPT="-U -B -T 1C -s $BUILD_DIR/.settings.xml ${{ inputs.verbose && '--debug' || '' }}"
           echo "::notice ::building - stream-applications-build"
           set -e
           ./mvnw $MAVEN_OPT install -DskipTests -f stream-applications-build
@@ -69,13 +69,12 @@ jobs:
         uses: tj-actions/changed-files@v34
       - name: 'Action: verify changed modules'
         run: |
-          $ROOT_DIR=$(realpath $PWD)
-          pushd stream-applications > /dev/null
-            $ROOT_DIR/create-matrices.sh
-            $ROOT_DIR/scripts/determine-modules-changed.sh "${{ steps.changed-files.outputs.all_changed_files }}"
-            MODULES=$(jq '.[]' modules.json | sed 's/\"//g')
-            for module in $MODULES; do
-              echo "::info ::verify module $module"
-              ./mvnw $MAVEN_OPT verify -f $module -amd
-            done
-          popd > /dev/null
+          BUILD_DIR=./main-build
+          $BUILD_DIR/create-matrices.sh
+          $BUILD_DIR/scripts/determine-modules-changed.sh "${{ steps.changed-files.outputs.all_changed_files }}"
+          MODULES=$(jq '.[]' modules.json | sed 's/\"//g')
+          for module in $MODULES; do
+            echo "::info ::verify module $module"
+            ./mvnw $MAVEN_OPT verify -f $module -amd
+          done
+          

--- a/functions/function/http-request-function/src/main/java/org/springframework/cloud/fn/http/request/HttpRequestFunctionConfiguration.java
+++ b/functions/function/http-request-function/src/main/java/org/springframework/cloud/fn/http/request/HttpRequestFunctionConfiguration.java
@@ -20,8 +20,6 @@ import java.time.Duration;
 import java.util.Map;
 import java.util.function.Function;
 
-import reactor.core.publisher.Flux;
-
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
@@ -42,7 +40,7 @@ import org.springframework.web.util.UriBuilderFactory;
  *
  * @author David Turanski
  * @author Sunny Hemdev
- *
+ * @author Corneil du Plessis
  **/
 @Configuration(proxyBeanMethods = false)
 @EnableConfigurationProperties(HttpRequestFunctionProperties.class)
@@ -63,7 +61,7 @@ public class HttpRequestFunctionConfiguration {
 	 * Function that accepts a {@code Flux<Message<?>>} containing body and headers and
 	 * returns a {@code Flux<ResponseEntity<?>>}.
 	 */
-	public static class HttpRequestFunction implements Function<Flux<Message<?>>, Flux<?>> {
+	public static class HttpRequestFunction implements Function<Message<?>, Object> {
 
 		private final WebClient webClient;
 
@@ -77,16 +75,17 @@ public class HttpRequestFunctionConfiguration {
 		}
 
 		@Override
-		public Flux<?> apply(Flux<Message<?>> messageFlux) {
-			return messageFlux.flatMap(message -> this.webClient
-					.method(resolveHttpMethod(message))
-					.uri(uriBuilderFactory.uriString(resolveUrl(message)).build())
-					.bodyValue(resolveBody(message))
-					.headers(httpHeaders -> httpHeaders.addAll(resolveHeaders(message)))
-					.retrieve()
-					.toEntity(properties.getExpectedResponseType())
-					.map(responseEntity -> properties.getReplyExpression().getValue(responseEntity))
-					.timeout(Duration.ofMillis(properties.getTimeout())));
+		public Object apply(Message<?> message) {
+			return this.webClient
+				.method(resolveHttpMethod(message))
+				.uri(uriBuilderFactory.uriString(resolveUrl(message)).build())
+				.bodyValue(resolveBody(message))
+				.headers(httpHeaders -> httpHeaders.addAll(resolveHeaders(message)))
+				.retrieve()
+				.toEntity(properties.getExpectedResponseType())
+				.map(responseEntity -> properties.getReplyExpression().getValue(responseEntity))
+				.timeout(Duration.ofMillis(properties.getTimeout()))
+				.block();
 		}
 
 		private String resolveUrl(Message<?> message) {
@@ -99,7 +98,7 @@ public class HttpRequestFunctionConfiguration {
 
 		private Object resolveBody(Message<?> message) {
 			return properties.getBodyExpression() != null ? properties.getBodyExpression().getValue(message)
-					: message.getPayload();
+				: message.getPayload();
 		}
 
 		private HttpHeaders resolveHeaders(Message<?> message) {
@@ -109,7 +108,7 @@ public class HttpRequestFunctionConfiguration {
 				for (Map.Entry<?, ?> header : headersMap.entrySet()) {
 					if (header.getKey() != null && header.getValue() != null) {
 						headers.add(header.getKey().toString(),
-								header.getValue().toString());
+							header.getValue().toString());
 					}
 				}
 			}

--- a/functions/function/http-request-function/src/test/java/org/springframework/cloud/fn/http/request/HttpRequestFunctionTestApplicationTests.java
+++ b/functions/function/http-request-function/src/test/java/org/springframework/cloud/fn/http/request/HttpRequestFunctionTestApplicationTests.java
@@ -27,8 +27,6 @@ import okhttp3.mockwebserver.MockWebServer;
 import okhttp3.mockwebserver.RecordedRequest;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import reactor.core.publisher.Flux;
-import reactor.test.StepVerifier;
 
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.test.context.runner.ApplicationContextRunner;
@@ -42,6 +40,7 @@ import org.springframework.messaging.support.GenericMessage;
 import org.springframework.messaging.support.MessageBuilder;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
 
 public class HttpRequestFunctionTestApplicationTests {
 	private MockWebServer server = new MockWebServer();
@@ -67,14 +66,9 @@ public class HttpRequestFunctionTestApplicationTests {
 		runner.withPropertyValues("http.request.http-method-expression='POST'").run(context -> {
 			HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
 			Message<?> message = MessageBuilder.withPayload("").build();
-			StepVerifier.create(httpRequestFunction.apply(Flux.just(message)))
-					.assertNext(o -> {
-						ResponseEntity r = (ResponseEntity) o;
-						assertThat(r.getBody()).isEqualTo("hello");
-						assertThat(r.getStatusCode().is2xxSuccessful()).isTrue();
-					})
-					.expectComplete()
-					.verify();
+			ResponseEntity r = (ResponseEntity) httpRequestFunction.apply(message);
+			assertThat(r.getBody()).isEqualTo("hello");
+			assertThat(r.getStatusCode().is2xxSuccessful()).isTrue();
 		});
 	}
 
@@ -86,29 +80,24 @@ public class HttpRequestFunctionTestApplicationTests {
 			public MockResponse dispatch(RecordedRequest recordedRequest) {
 				return new MockResponse().setHeader(HttpHeaders.CONTENT_TYPE,
 						recordedRequest.getHeader(HttpHeaders.CONTENT_TYPE))
-						.setBody(recordedRequest.getBody())
-						.setResponseCode(HttpStatus.CREATED.value());
+					.setBody(recordedRequest.getBody())
+					.setResponseCode(HttpStatus.CREATED.value());
 			}
 		});
 
 		runner.withPropertyValues("http.request.http-method-expression='POST'",
-				"http.request.headers-expression={'Content-Type':'application/json'}").run(context -> {
-					HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
-					String json = "{\"hello\":\"world\"}";
-					Message<?> message = MessageBuilder.withPayload(json)
-							.build();
-					StepVerifier.create(httpRequestFunction.apply(Flux.just(message)))
-							.assertNext(o -> {
-								ResponseEntity r = (ResponseEntity) o;
-								assertThat(r.getBody()).isEqualTo(json);
-								assertThat(r.getStatusCode().is2xxSuccessful()).isTrue();
-								assertThat(r.getHeaders().getContentType()).isEqualTo(MediaType.APPLICATION_JSON);
-							})
-							.expectComplete()
-							.verify();
-					RecordedRequest request = server.takeRequest(100, TimeUnit.MILLISECONDS);
-					assertThat(request.getMethod()).isEqualTo("POST");
-				});
+			"http.request.headers-expression={'Content-Type':'application/json'}").run(context -> {
+			HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
+			String json = "{\"hello\":\"world\"}";
+			Message<?> message = MessageBuilder.withPayload(json)
+				.build();
+			ResponseEntity r = (ResponseEntity) httpRequestFunction.apply(message);
+			assertThat(r.getBody()).isEqualTo(json);
+			assertThat(r.getStatusCode().is2xxSuccessful()).isTrue();
+			assertThat(r.getHeaders().getContentType()).isEqualTo(MediaType.APPLICATION_JSON);
+			RecordedRequest request = server.takeRequest(100, TimeUnit.MILLISECONDS);
+			assertThat(request.getMethod()).isEqualTo("POST");
+		});
 	}
 
 	@Test
@@ -119,29 +108,24 @@ public class HttpRequestFunctionTestApplicationTests {
 			public MockResponse dispatch(RecordedRequest recordedRequest) {
 				return new MockResponse().setHeader(HttpHeaders.CONTENT_TYPE,
 						recordedRequest.getHeader(HttpHeaders.CONTENT_TYPE))
-						.setBody(recordedRequest.getBody())
-						.setResponseCode(HttpStatus.CREATED.value());
+					.setBody(recordedRequest.getBody())
+					.setResponseCode(HttpStatus.CREATED.value());
 			}
 		});
 
 		runner.withPropertyValues("http.request.http-method-expression='POST'",
-				"http.request.expected-response-type=" + Map.class.getName()).run(context -> {
-					HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
-					Map<String, String> json = Collections.singletonMap("hello", "world");
-					Message<?> message = MessageBuilder.withPayload(json)
-							.build();
-					StepVerifier.create(httpRequestFunction.apply(Flux.just(message)))
-							.assertNext(o -> {
-								ResponseEntity r = (ResponseEntity) o;
-								assertThat(r.getBody()).isEqualTo(json);
-								assertThat(r.getStatusCode().is2xxSuccessful()).isTrue();
-								assertThat(r.getHeaders().getContentType()).isEqualTo(MediaType.APPLICATION_JSON);
-							})
-							.expectComplete()
-							.verify();
-					RecordedRequest request = server.takeRequest(100, TimeUnit.MILLISECONDS);
-					assertThat(request.getMethod()).isEqualTo("POST");
-				});
+			"http.request.expected-response-type=" + Map.class.getName()).run(context -> {
+			HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
+			Map<String, String> json = Collections.singletonMap("hello", "world");
+			Message<?> message = MessageBuilder.withPayload(json)
+				.build();
+			ResponseEntity r = (ResponseEntity) httpRequestFunction.apply(message);
+			assertThat(r.getBody()).isEqualTo(json);
+			assertThat(r.getStatusCode().is2xxSuccessful()).isTrue();
+			assertThat(r.getHeaders().getContentType()).isEqualTo(MediaType.APPLICATION_JSON);
+			RecordedRequest request = server.takeRequest(100, TimeUnit.MILLISECONDS);
+			assertThat(request.getMethod()).isEqualTo("POST");
+		});
 	}
 
 	@Test
@@ -157,21 +141,16 @@ public class HttpRequestFunctionTestApplicationTests {
 		});
 
 		runner.withPropertyValues("http.request.http-method-expression='DELETE'",
-				"http.request.expected-response-type=" + Void.class.getName()).run(context -> {
-					HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
-					Message<?> message = MessageBuilder.withPayload("")
-							.build();
-					StepVerifier.create(httpRequestFunction.apply(Flux.just(message)))
-							.assertNext(o -> {
-								ResponseEntity r = (ResponseEntity) o;
-								assertThat(r.getBody()).isNull();
-								assertThat(r.getStatusCode().is2xxSuccessful()).isTrue();
-							})
-							.expectComplete()
-							.verify();
-					RecordedRequest request = server.takeRequest(100, TimeUnit.MILLISECONDS);
-					assertThat(request.getMethod()).isEqualTo("DELETE");
-				});
+			"http.request.expected-response-type=" + Void.class.getName()).run(context -> {
+			HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
+			Message<?> message = MessageBuilder.withPayload("")
+				.build();
+			ResponseEntity r = (ResponseEntity) httpRequestFunction.apply(message);
+			assertThat(r.getBody()).isNull();
+			assertThat(r.getStatusCode().is2xxSuccessful()).isTrue();
+			RecordedRequest request = server.takeRequest(100, TimeUnit.MILLISECONDS);
+			assertThat(request.getMethod()).isEqualTo("DELETE");
+		});
 	}
 
 	@Test
@@ -179,9 +158,12 @@ public class HttpRequestFunctionTestApplicationTests {
 		server.shutdown();
 		runner.run(context -> {
 			HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
-			StepVerifier.create(httpRequestFunction.apply(Flux.just(new GenericMessage(""))))
-					.expectErrorMatches(throwable -> throwable.getMessage().contains("Connection refused"))
-					.verify();
+			try {
+				httpRequestFunction.apply(new GenericMessage(""));
+				fail("Expected exception");
+			} catch (Throwable throwable) {
+				assertThat(throwable.getMessage()).contains("Connection refused");
+			}
 		});
 	}
 
@@ -202,25 +184,20 @@ public class HttpRequestFunctionTestApplicationTests {
 				"http.request.http-method-expression=headers['method']",
 				"http.request.body-expression=headers['body']",
 				"http.request.headers-expression={Accept:'application/json'}")
-				.run(context -> {
-					Message<?> message = MessageBuilder.withPayload("")
-							.setHeader("url", url())
-							.setHeader("method", "POST")
-							.setHeader("body", "{\"hello\":\"world\"}")
-							.build();
+			.run(context -> {
+				Message<?> message = MessageBuilder.withPayload("")
+					.setHeader("url", url())
+					.setHeader("method", "POST")
+					.setHeader("body", "{\"hello\":\"world\"}")
+					.build();
 
-					HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
-					StepVerifier.create(httpRequestFunction.apply(Flux.just(message)))
-							.assertNext(o -> {
-								ResponseEntity responseEntity = (ResponseEntity) o;
-								assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
-								assertThat(responseEntity.getBody()).isEqualTo(message.getHeaders().get("body"));
-								assertThat(responseEntity.getHeaders().getContentType())
-										.isEqualTo(MediaType.APPLICATION_JSON);
-							})
-							.expectComplete()
-							.verify();
-				});
+				HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
+				ResponseEntity responseEntity = (ResponseEntity) httpRequestFunction.apply(message);
+				assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+				assertThat(responseEntity.getBody()).isEqualTo(message.getHeaders().get("body"));
+				assertThat(responseEntity.getHeaders().getContentType())
+					.isEqualTo(MediaType.APPLICATION_JSON);
+			});
 	}
 
 	@Test
@@ -238,21 +215,16 @@ public class HttpRequestFunctionTestApplicationTests {
 				"http.request..http-method-expression='POST'",
 				"http.request.headers-expression={'Content-Type':'application/octet-stream'}",
 				"http.request.expected-response-type=byte[]")
-				.run(context -> {
-					Message<?> message = MessageBuilder.withPayload("hello")
-							.build();
-					HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
-					StepVerifier.create(httpRequestFunction.apply(Flux.just(message)))
-							.assertNext(o -> {
-								ResponseEntity responseEntity = (ResponseEntity) o;
-								assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
-								assertThat(responseEntity.getBody()).isEqualTo("hello".getBytes());
-								assertThat(responseEntity.getHeaders().getContentType())
-										.isEqualTo(MediaType.APPLICATION_OCTET_STREAM);
-							})
-							.expectComplete()
-							.verify();
-				});
+			.run(context -> {
+				Message<?> message = MessageBuilder.withPayload("hello")
+					.build();
+				HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
+				ResponseEntity responseEntity = (ResponseEntity) httpRequestFunction.apply(message);
+				assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+				assertThat(responseEntity.getBody()).isEqualTo("hello".getBytes());
+				assertThat(responseEntity.getHeaders().getContentType())
+					.isEqualTo(MediaType.APPLICATION_OCTET_STREAM);
+			});
 	}
 
 	@Test
@@ -268,21 +240,15 @@ public class HttpRequestFunctionTestApplicationTests {
 		});
 		runner.withPropertyValues(
 				"http.request.http-method-expression=#jsonPath(payload,'$.myMethod')")
-				.run(context -> {
-					Message<?> message = MessageBuilder
-							.withPayload("{\"name\":\"Fred\",\"age\":41, \"myMethod\":\"POST\"}")
-							.build();
-					HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
-					StepVerifier.create(httpRequestFunction.apply(Flux.just(message)))
-							.assertNext(o -> {
-								ResponseEntity responseEntity = (ResponseEntity) o;
-								assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
-								assertThat(responseEntity.getBody()).isEqualTo(message.getPayload());
-							})
-							.expectComplete()
-							.verify();
-
-				});
+			.run(context -> {
+				Message<?> message = MessageBuilder
+					.withPayload("{\"name\":\"Fred\",\"age\":41, \"myMethod\":\"POST\"}")
+					.build();
+				HttpRequestFunction httpRequestFunction = context.getBean(HttpRequestFunction.class);
+				ResponseEntity responseEntity = (ResponseEntity) httpRequestFunction.apply(message);
+				assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+				assertThat(responseEntity.getBody()).isEqualTo(message.getPayload());
+			});
 	}
 
 	private String url() {

